x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityRCS1FailureStoreRestIT.java

```diff
@@ -59,10 +69,190 @@ public class RemoteClusterSecurityRCS1FailureStoreRestIT extends AbstractRemoteC
     private static final String BACKING_FAILURE_STORE_INDEX_ACCESS = "backing_failure_store_index_access";
     private static final String BACKING_DATA_INDEX_ACCESS = "backing_data_index_access";
 
+    /**
+     * Maps usernames to their role descriptors. The usernames are also used as role names.
+     */
+    private static final Map<String, String> usersAndRolesOnFulfillingCluster = Map.of(DATA_ACCESS, """
+        {
+          "indices": [
+            {
+              "names": ["test*"],
+              "privileges": ["read", "read_cross_cluster"]
+            }
+          ]
+        }""", FAILURE_STORE_ACCESS, """
+        {
+          "indices": [
+            {
+              "names": ["test*", "non-existing-index"],
+              "privileges": ["read", "read_cross_cluster", "read_failure_store"]
+            }
+          ]
+        }""", MANAGE_FAILURE_STORE_ACCESS, """
+        {
+          "indices": [
+            {
+              "names": ["test*", "non-existing-index"],
+              "privileges": ["manage_failure_store", "read_cross_cluster", "read_failure_store"]
+            }
+          ]
+        }""", ALL_ACCESS, """
+        {
+          "indices": [
+            {
+              "names": ["*"],
+              "privileges": ["all"]
+            }
+          ]
+        }""", ONLY_READ_FAILURE_STORE_ACCESS, """
+        {
+          "indices": [
+            {
+              "names": ["test*", "non-existing-index"],
+              "privileges": ["read_failure_store"]
+            }
+          ]
+        }""", BACKING_DATA_INDEX_ACCESS, """
+        {
+          "indices": [
+            {
+              "names": [".ds-test*"],
+              "privileges": ["read", "read_cross_cluster"]
+            }
+          ]
+        }""", BACKING_FAILURE_STORE_INDEX_ACCESS, """
+        {
+          "indices": [
+            {
+              "names": [".fs-test*"],
+              "privileges": ["read", "read_cross_cluster"]
+            }
+          ]
+        }""");
+
+    /**
+     * The role must simply exist on query cluster, the actual access is irrelevant for user authorization.
+     * But we here grant access to the local and (for some also) remote indices to test mixed search
+     * and API key authorization.
+     */
+    private static final Map<String, String> usersAndRolesOnQueryCluster = Map.of(DATA_ACCESS, """
+        {
+          "cluster": ["manage_security"],
+          "indices": [
+             {
+              "names": ["local_index"],
+              "privileges": ["read"]
+            },
+            {
```

Comment:
The roles on query cluster now include index privileges which were previously only defined on fulfilling cluster. This is needed, because API keys are created on query cluster for each user. Without this, API keys would be useless as they would not have any privilege to access because they are limited-by owning user's privileges.